Account for scimilarity in cell typing methods (#201)
AlexsLemonade/ScPCA-manuscript: content/04.methods.md

For each cell, `CellAssign` calculates a probability of assignment to each cell type in the reference.
The probability matrix and a prediction based on the most probable cell type were added as cell type annotations to the processed `SingleCellExperiment` object.

For `SCimilarity`, the foundation model described in Heimberg et al. [@doi:10.1038/s41586-024-08411-y], containing 7.3 million cells from various normal and diseased tissues was obtained from zenodo (<https://zenodo.org/records/10685499>) and used to annotate cells in all samples.

> The assigned cell type label and the minimum distance of the query cell to a reference cell in the model were added to the processed `SingleCellExperiment` object.

Reviewer: This phrasing seems off to me, specifically "to a reference cell."

The [tutorial](https://genentech.github.io/scimilarity/notebooks/cell_annotation_tutorial.html#Annotation-QC) says:
> `min_dist`, represents the minimum distance between a cell in the query dataset and all cells in the training set

So, instead of "a reference cell"...eh? Or am I really wrong here?

```suggestion
The assigned cell type label and the minimum distance of the query cell to reference cells of the same label in the model were added to the processed `SingleCellExperiment` object.
```

#### Assigning consensus cell types

Cell type labels obtained from `SingleR`, `CellAssign`, and `SCimilarity` were then used to assign an ontology-aware consensus cell type label.
We first assigned each of the cell types present in the `PanglaoDB` [@doi:10.1093/database/baz046] reference used with `CellAssign` to an appropriate Cell Ontology term [@url:https://www.ebi.ac.uk/ols4/ontologies/cl].
For cell types available in the `BlueprintEncodeData` reference used with `SingleR` and the foundation model used with `SCimilarity`, we used the provided Cell Ontology terms.

We then created a reference table containing all possible combinations of cell types assigned using `SingleR`, `CellAssign`, and `SCimilarity`. 
Consensus cell types are assigned if two of the three annotations share a latest common ancestor (LCA), identified using `ontoProc::findCommonAncestors()` [@doi:10.18129/B9.bioc.ontoProc], that meets the following critera. 
Otherwise, no consensus cell type is assigned and the cell is labeled as "Unknown".

> 1. The terms share at least 1 LCA with fewer than 170 descendants, with the exception of `neuron`, `epithelial cell`, `columnar/cuboidal epithelial cell` or `endo-epithelial cell`.

Reviewer: So I realize this is the same as what we already added to `scpca-docs`, but reading it fresh now I find it confusing. The _exception_ refers to no consensus cell type being assigned - a consensus cell type is assigned if it's one of these which have more than 170. 

I think this will help - 

```suggestion
1. The terms share at least 1 LCA that either has fewer than 170 descendants or is one of `neuron`, `epithelial cell`, `columnar/cuboidal epithelial cell` or `endo-epithelial cell`.
```

What do you think? If you like it, we can also update in docs.
